### PR TITLE
FIX-430: compilation errors and warnings

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
Not sure if I fixed everything properly, please double check

### CODE REVIEW

File: include/eve/arch/x86/as_register.hpp [hunk 1]
```diff
@@ -19,6 +19,9 @@ namespace eve
   struct logical;
   struct x86_128_;
   struct x86_256_;
+
+  template<typename Type, typename Cardinal, typename ABI>
+  struct as_register;
```
Comment (DenisYaroshevskiy) [inline include/eve/arch/x86/as_register.hpp:24]: Missing `as_register` primary definition. Should I have included smth instead?

File: include/eve/concept/memory.hpp [hunk 2]
```diff
@@ -27,7 +27,7 @@ namespace eve
 
   template<typename T, typename SIMD>
   concept simd_compatible_ptr = simd_value<SIMD> && (   std::is_pointer_v<T>
-                                                    ||  detail::is_aligned_ptr<T>::value
-                                                    && (pointer_alignment_v<T> >= alignment_v<SIMD>)
+                                                    ||  (detail::is_aligned_ptr<T>::value
+                                                         && (pointer_alignment_v<T> >= alignment_v<SIMD>))
```
Comment (DenisYaroshevskiy) [inline include/eve/concept/memory.hpp:31]: || and && mixing without brackets warning

File: include/eve/conditional.hpp [hunk 3]
```diff
@@ -11,10 +11,11 @@
 #pragma once
 
 #include <eve/as.hpp>
+#include <eve/assert.hpp>
 #include <eve/concept/conditional.hpp>
 #include <eve/detail/abi.hpp>
 #include <eve/detail/function/iota.hpp>
-#include <eve/assert.hpp>
+#include <eve/function/bit_cast.hpp>
```
Comment (DenisYaroshevskiy) [inline include/eve/conditional.hpp:18]: Missing bit_cast declaration

File: include/eve/module/real/core/function/regular/simd/x86/is_less.hpp [hunk 4]
```diff
@@ -78,7 +80,6 @@ namespace eve::detail
       else
       {
         constexpr auto use_avx2 = current_api >= avx2;
-        constexpr auto use_sse4 = current_api >= sse4_2;
```
Comment (DenisYaroshevskiy) [inline include/eve/module/real/core/function/regular/simd/x86/is_less.hpp:81]: unused

File: include/eve/arch/x86/as_register.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/arch/x86/as_register.hpp:24]: It's normally in the arch/cpu/as_register.hpp
Forward declaring is fine too.

Comment (jfalcou) [reply include/eve/arch/x86/as_register.hpp:24]: But it doesn't break any of our compilers even locally. The as_register should be used via the main arch/as_register.hpp as it includes everything it requires.

File: include/eve/concept/memory.hpp (hunk 2 above)
Comment (jfalcou) [reply include/eve/concept/memory.hpp:31]: I got this warning once but never had to pop it back.

**Review by jfalcou — APPROVED** [review]

**Pull request merged**